src/components/navigation.tsx

```diff
@@ -7,76 +8,166 @@ import {
   NavigationMenuList,
   NavigationMenuTrigger,
 } from "@/components/ui/navigation-menu";
+import {
+  RiMenu2Fill,
+  RiCloseFill,
+  RiArrowUpSLine,
+  RiArrowDownSLine,
+} from "react-icons/ri";
+import navBarItems from "@/data/navBarData";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
 import Link from "next/link";
+
 const Navigation = () => {
+  const [navBar, setNav] = useState(false);
+  const [programBar, setProgramBar] = useState(false);
+  const router = useRouter();
+
+  const handleNavBar = () => {
+    setNav(!navBar);
+  };
+
+  const handleProgramBar = () => {
+    setProgramBar(!programBar);
+  };
+
+  const handleBothBars = () => {
+    setProgramBar(!programBar);
+    setNav(!navBar);
+  };
+
   return (
-    <div className="border-acm-blue-400 grid grid-cols-5 border-b-3 p-5 shadow-xl">
-      <Link className="text-acm-gray-500 text-2xl font-bold" href="/">
+    <div className="border-acm-blue-400 grid grid-cols-4 border-b-3 p-5 shadow-xl md:grid-cols-5">
+      <RiMenu2Fill
+        className="text-acm-gray-500 block md:hidden"
+        size={32}
+        onClick={programBar ? handleBothBars : handleNavBar}
+      />
+      <Link
+        className="text-acm-gray-500 pl-4 text-2xl font-bold md:pl-0"
```

Comment:
```suggestion
        className="text-acm-gray-500 pl-4 text-3xl font-bold md:pl-0"
```
lowk let's make this text larger lmao